### PR TITLE
Method testing

### PR DESCRIPTION
**Pull request by KerronKing**

### CODE REVIEW

File: spec/main_spec.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require './lib/main'
+
+RSpec.describe Enumerable do
+  describe '#my_count' do
+    it 'calls a block on an object and returns the number of elements' do
+      arr = [1, 2, 3, 4, 5]
+      expect(arr.my_count).to eql(5)
```
Comment (bolah2009) [inline spec/main_spec.rb]: :thinking: What happens when a block is passed? Or an argument is passed?
Please, let's add edge cases for the test. This test is not yet effective as it passes even like this (Commenting out part of the code). :point_down: 
```ruby
def my_count(&block)
    # result = []
    return size unless block_given?

    # my_each do |elem|
    #   result << elem if block.call(elem) == true
    # end
    # result.size
  end
```

File: spec/main_spec.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      arr = [1, 2, 3]
+      result = [2, 4, 6]
+      expect(arr.my_map { |x| x * 2 }).to(eql(result))
+    end
+  end
+
+  describe '#my_any?' do
+    it 'returns true if the block passed returns true for any item in the array' do
+      arr = [1, 2, 3]
+      expect(arr.my_any? { |x| x > 2 }).to(eql(true))
```
Comment (bolah2009) [inline spec/main_spec.rb]: :thinking: What happens when the block returns false for each value? For example: :point_down: 
```suggestion
      expect(arr.my_any? { |x| x > 2 }).to(eql(true))
      expect(arr.my_any? { |x| x > 5 }).to(eql(false))
```
Please, let's add edge cases for the test. This test is not yet effective as it passes even like this (Commenting out part of the code). :point_down: 
```ruby
  def my_any?(&block)
    counter = 0
    my_each do |elem|
      counter += 1 if block.call(elem) == true
    end
    return true if counter.positive?

    # false
  end
```

File: spec/main_spec.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    end
+  end
+
+  describe '#my_none?' do
+    it 'returns true if the block passed returns false for each item in the array' do
+      arr = [1, 2, 3]
+      expect(arr.my_none? { |x| x > 4 }).to(eql(true))
```
Comment (bolah2009) [inline spec/main_spec.rb]: Check comment above on writing both negative and positive cases.

File: spec/main_spec.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  describe '#my_count' do
+    it 'calls a block on an object and returns the number of elements' do
+      arr = [1, 2, 3, 4, 5]
+      expect(arr.my_count).to eql(5)
+    end
+  end
+
+  describe '#my_map' do
+    it 'returns a new array as a result of calling a block once on each item in the original array' do
+      arr = [1, 2, 3]
```
Comment (bolah2009) [inline spec/main_spec.rb]: #### `REQUIRED`
Let's use `let` here instead. I can see you use `arr = [1, 2, 3]` in four different places already. You can define this `arr` using `let` helper method after the first `describe` block and use it anywhere else. See example below:

```ruby
RSpec.describe Enumerable do
let(:arr){[1, 2, 3]}

```

Learn more about [`let` here](https://relishapp.com/rspec/rspec-core/v/2-5/docs/helper-methods/let-and-let)

File: spec/main_spec.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      result = [2, 4, 6]
+      expect(arr.my_map { |x| x * 2 }).to(eql(result))
```
Comment (bolah2009) [inline spec/main_spec.rb]: #### `OPTIONAL` 
You can also use `to` without the parenthesis.

```ruby
expect(arr.my_map { |x| x * 2 }).to eql(result)
```

**Review by bolah2009 — CHANGES_REQUESTED** [review]
# Hi @KerronKing    :wave: 

### Awesome Job :muscle: 

Few things before getting a :+1: 
- Add Descriptive readme
  > Remember about descriptive README file.
- Is there a reason why you using a new repo? :thinking: 
  > Use existing repositories for `Tic Tac Toe` and `Enumerables`: add specs and open new Pull Requests.
- Test for both negative and positive cases. See comment below.
  > Use your common sense: test positive and negative scenarios.
- Use `let` to define variables for test cases
- Add PR description

| Current | Preffred |
|:-----:|:-----|
| prking | update-PR-description-html |

#### Is there a reason you the only one working on this project? This is a **collaborative** project. :thinking: 

#### Fix this and submit another code review request. :ok_hand: 

#### Feel free to reach me on slack if you having issues implementing these features. :+1: 

Good luck with your next submission. :star2: 

Cheers :confetti_ball: 
[Bola](https://bolabuari.com/)

###### PS: Note that comments tagged as `OPTIONAL` are not part of the minimum requirements to approve this PR

File: spec/main_spec.rb [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    end
+
+    it 'returns the number of items in the object that are equal to the argument given' do
+      expect(arr.my_count(2)).to eql(1)
+    end
+
+    it 'calls a block on an object and returns the number of elements that evaluate to true' do
+      expect(arr.my_count { |x| x < 4 }).to eql(3)
+    end
+  end
```
Comment (samgaco) [inline spec/main_spec.rb:20]: Block and argument test cases implemented :+1:

File: spec/main_spec.rb [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+      expect(arr.my_map { |x| x - 2 }).to eql(result)
+    end
+  end
+
+  describe '#my_any?' do
+    it 'returns true if the block passed returns true for any item in the array' do
+      expect(arr.my_any? { |x| x > 2 }).to eql(true)
+    end
+    it 'returns false if the block passed returns false for all item in the array' do
+      expect(arr.my_any? { |x| x > 3 }).to eql(false)
```
Comment (samgaco) [inline spec/main_spec.rb:44]: :100:

File: spec/main_spec.rb [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    end
+  end
+
+  describe '#my_none?' do
+    it 'returns true if the block passed returns false for each item in the array' do
+      expect(arr.my_none? { |x| x > 4 }).to eql(true)
+    end
+    it 'returns false if the block passed returns true for any item in the array' do
+      expect(arr.my_none? { |x| x > 2 }).to eql(false)
```
Comment (samgaco) [inline spec/main_spec.rb:53]: :white_check_mark:

**Review by samgaco — APPROVED** [review]
#### Hi @KerronKing and @MarvellousUbani :high_brightness: 

#### All the requested changes have been met therefore the project is approved :white_check_mark: . You may now merge to the master branch and submit the project completion form :1st_place_medal: 

------------------------------
**[SUGGESTION]** I would suggest improving the readme file, you might check this [template](https://github.com/othneildrew/Best-README-Template) for inspiration or this [course](https://www.udacity.com/course/writing-readmes--ud777). You could try to answer questions that someone going into your repo not familiar with the project would ask himself.

--------------------------------
* ###### STATUS OF THE PROJECT: APPROVED :heavy_check_mark:
